pkg/kubelet/kubelet_node_status.go

```diff
@@ -948,7 +949,7 @@ var oldNodeUnschedulable bool
 // record if node schedulable change.
 func (kl *Kubelet) recordNodeSchedulableEvent(node *v1.Node) {
 	if oldNodeUnschedulable != node.Spec.Unschedulable {
-		if node.Spec.Unschedulable {
+		if node.Spec.Unschedulable || taints.TaintExistsWithEffect(node.Spec.Taints) {
```

Comment:
If you want to do that you need to update the previous `if` as well. But generally I don't think this Event is particularly useful for Taints, so I'd just drop it and add a comment that we chose to ignore checking Taint here.

Reply:
@gmarek - I'd update the upper if, at least we'll have a uniform behaviour across the code base, is it fine for you ?

Reply:
We shouldn't emit Event when the Taint is added to the Node. See https://github.com/kubernetes/community/blob/master/contributors/devel/event-style-guide.md.